src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializer.java

```diff
@@ -230,6 +230,11 @@ private void validate(
                   ErrorCode.INVALID_FILTER_EXPRESSION,
                   "$size operator must have integer value >= 0");
             }
+            // Check if the value is an integer by comparing its scale.
+            if (i.stripTrailingZeros().scale() > 0) {
```

Comment:
I don't think we have suitable `NumberUtil` class yet, but in future we might want to create something like `ValueUtil` to collect helper methods like this one here (I did check and this is indeed a good way for checking).

Reply:
Thanks